### PR TITLE
UI for specifying default behaviour for apps

### LINKED ISSUE #8: Specify default browsers for particular apps, windows, etc.

### PR DESCRIPTION
**Pull request by artem-fomin-by**
Added opportunity to define default link opening behaviour for particular apps

### CODE REVIEW

**Comment by alex-fomin** [comment]
Please add screenshots here

File: Logic/Config.cs [hunk 1]
```diff
@@ -17,4 +19,14 @@ public class Config
 
         return null;
     }
+
+    public void Deserialize(string configurationFilePath)
+    {
+        using var configurationFile = File.Open(configurationFilePath, FileMode.Open, FileAccess.Write);
+        JsonSerializer.Serialize(
```
Comment (alex-fomin) [inline Logic/Config.cs:26]: Method is called `Deserialize`, but you are serializing - where is a mistake?

File: WPFMainApp/MainWindow.xaml.cs [hunk 2]
```diff
@@ -62,16 +61,21 @@ private void AdjustInitPosition()
...
     }
 
-    private void InitButtons(Browser[] browsers, double maxWidth, double maxHeight)
+    private void InitControls(Browser[] browsers, double maxWidth, double maxHeight)
     {
+        DefaultBrowserCheckBox.Content = string.Format(
+            "Always open links from \"{0}\" in selected browser", 
+            App.ParentProcess?.ProcessName ?? ""
+            );
+        DefaultBrowserCheckBox.Margin = new Thickness(2.0, 2.0, 2.0, 2.0);
```
Comment (alex-fomin) [inline WPFMainApp/MainWindow.xaml.cs:77]: As far as margins are constantly here, plz move it to layout.

File: WPFMainApp/App.cs [hunk 3]
```diff
@@ -85,4 +89,20 @@ private static void EnsureConfiguration(string configurationFilePath, string con
             }, new JsonSerializerOptions { WriteIndented = true });
         }
     }
+
+    public void End(Browser selectedBrowser)
```
Comment (alex-fomin) [inline WPFMainApp/App.cs:93]: Not a fortunate name for the method. Do not call `Shutdown` here, just save settings to file.

File: WPFMainApp/MainWindow.xaml.cs [hunk 4]
```diff
@@ -89,11 +93,14 @@ private void InitButtons(Browser[] browsers, double maxWidth, double maxHeight)
...
                 browser.Launch(_link);
-                _parentApp.Shutdown();
+                if(DefaultBrowserCheckBox.IsChecked == true) _parentApp.End(browser);
+                else _parentApp.Shutdown();
             };
 
             ButtonsStackPanel.Children.Add(button);
         }
+
+        if(App.ParentProcess == null) ((Grid)Content).Children.RemoveAt(2);
```
Comment (alex-fomin) [inline WPFMainApp/MainWindow.xaml.cs:103]: what a magic number here?

**Review by alex-fomin — CHANGES_REQUESTED** [review]

File: WPFMainApp/MainWindow.xaml.cs (hunk 2 above)
Comment (artem-fomin-by) [reply WPFMainApp/MainWindow.xaml.cs]: Resolved in the recent commit

File: WPFMainApp/App.cs (hunk 3 above)
Comment (artem-fomin-by) [reply WPFMainApp/App.cs]: Resolved in the recent commit

File: WPFMainApp/MainWindow.xaml.cs (hunk 4 above)
Comment (artem-fomin-by) [reply WPFMainApp/MainWindow.xaml.cs]: Resolved in the recent commit

File: Logic/Config.cs (hunk 1 above)
Comment (artem-fomin-by) [reply Logic/Config.cs:26]: Resolved in the recent commit

**Pull request closed without merging**